### PR TITLE
Fix updating rich text via nested attributes

### LINKED ISSUE #35159: Action Text: Unable to consistently update has_rich_text field using accepts nested attributes
### Steps to reproduce
I had trouble creating a one-file script to support loading ActionText, so I've created a barebones repository here: https://github.com/mike1o1/action-text-test with passing/failing tests in [message_test.rb](https://github.com/mike1o1/action-text-test/blob/2217a264fb0e68142f8bfec8294437467d820ad3/test/models/message_test.rb)

It seems that if I have a model with a `has_one/has_many` relationship to an item that has rich text, then if I attempt to update only the rich text using `accepts_nested_attributes_for`, then the text is never updated. It seems as if the model doesn't realize there were changes, and so the write to the database never takes place.

If I update another attribute along with the rich text, then the update saves correctly.

Here is an example, with basic models:

```ruby
class Message < ApplicationRecord
    has_rich_text :content

	# also occurs with `has_many`
    has_one :post

    accepts_nested_attributes_for :post
end

class Post < ApplicationRecord
    belongs_to :message

    has_rich_text :content
end
```

And the corresponding unit test:

```ruby
test "can update post from within message" do
    message = Message.create!(title: "Seasons")
    post = message.create_post!

    # test updating post directly from the message
    message.update!(post_attributes: { content: "Test content", id: post.id })

    # it appears that everything updated?
    assert_equal "Test content", message.post.content.to_plain_text

    # let's reload the model and double check
    message.reload

    # this fails. The content is never saved. Uhoh
    assert_equal "Test content", message.post.content.to_plain_text
end
```

I'm not sure if I'm not using things correctly, but this is really subtle issue of things not working as I would have expected.

Even more confusingly, if I update another attribute on my `Post` model, then the rich text is saved correctly.

```ruby
# rich text is updated correctly
message.update!(post_attributes: { content: "Test content", id: post.id, title: "Plain text" })
```

### Expected behavior
I would expect that if I am updating a child object which contains rich text, then it would update correctly when using `accepts_nested_attributes_for`.

### Actual behavior
The rich text is not updated, unless *another* property on the model is also updated at the same time.

As a workaround, I save the child model directly if the `update` succeeds. This seems to force the database write.

### System configuration
**Rails version**: 6.0.0.beta1

**Ruby version**: 2.5.1, 2.6.0 and 2.6.1

### PR DESCRIPTION
**Pull request by georgeclaghorn**
Closes #35159.

### CODE REVIEW

File: actiontext/lib/action_text/attribute.rb [hunk 1]
```diff
@@ -34,14 +34,11 @@ def #{name}=(body)
           end
         CODE
 
-        has_one :"rich_text_#{name}", -> { where(name: name) }, class_name: "ActionText::RichText", as: :record, inverse_of: :record, dependent: :destroy
+        has_one :"rich_text_#{name}", -> { where(name: name) },
+          class_name: "ActionText::RichText", as: :record, inverse_of: :record, autosave: true, dependent: :destroy
```
Comment (georgeclaghorn) [inline actiontext/lib/action_text/attribute.rb]: Here’s the relevant bit: adding the `autosave: true` option to this association. This causes changes to the `ActionText::RichText`’s `body` attribute to be treated as autosaveable changes to the original record.

File: actiontext/test/unit/model_test.rb [hunk 2]
```diff
@@ -49,8 +49,16 @@ class ActionText::ModelTest < ActiveSupport::TestCase
...
     assert_equal "Hello world", message.body.to_plain_text
   end
+
+  test "updating content via nested attributes" do
+    message = Message.create!(subject: "Greetings", content: "<h1>Hello world</h1>")
+    message.create_review!(author_name: "Marcia", content: "Nice work!")
+
+    message.update!(review_attributes: { id: message.review.id, content: "Great work!" })
+    assert_equal "Great work!", message.review.reload.content.to_plain_text
+  end
```
Comment (kaspth) [inline actiontext/test/unit/model_test.rb]: Should we test creating via nested attributes as well?

File: actiontext/lib/action_text/attribute.rb [hunk 3]
```diff
@@ -34,14 +34,11 @@ def #{name}=(body)
...
 
         scope :"with_rich_text_#{name}", -> { includes("rich_text_#{name}") }
         scope :"with_rich_text_#{name}_and_embeds", -> { includes("rich_text_#{name}": { embeds_attachments: :blob }) }
-
-        after_save do
-          public_send(name).save if public_send(name).changed?
-        end
```
Comment (kaspth) [inline actiontext/lib/action_text/attribute.rb]: Neat!

**Review by kaspth — APPROVED** [review]

File: actiontext/test/unit/model_test.rb [hunk 4]
```diff
@@ -49,8 +49,22 @@ class ActionText::ModelTest < ActiveSupport::TestCase
...
   end
+
+  test "saving content via nested attributes" do
+    message = Message.create! subject: "Greetings", content: "<h1>Hello world</h1>",
+      review_attributes: { author_name: "Marcia", content: "Nice work!" }
+    assert_equal "Nice work!", message.review.content.to_plain_text
+  end
+
+  test "updating content via nested attributes" do
+    message = Message.create! subject: "Greetings", content: "<h1>Hello world</h1>",
```
Comment (abhaynikam) [inline actiontext/test/unit/model_test.rb]: Hi @georgeclaghorn ,

When I had debugged this issue, nested attributes update was already working when you pass other database columns(in this case `author_name` for `review_attributes `) in nested attributes hash. It was failing when only rich text field was updated via nested attributes. 

Something like this: https://github.com/rails/rails/pull/35430/files#diff-7706ccd89722c2baed5ee7c410e9f65bR71

I think we should add a test case for it. Let me know if I should open up a PR for it?

Comment (georgeclaghorn) [reply actiontext/test/unit/model_test.rb]: We’re not passing any other review attributes to `update`, and this test fails without the code changes in this PR.

**Pull request merged**